data/pigui/game.lua

```diff
@@ -0,0 +1,539 @@
+		else
+			return forward:angle() + adjust
+		end
+  end
+	local forward = Engine.ShipSpaceToScreenSpace(Vector(0,0,-1)) - center
+	local showDirection = displayDirectionalMarker(Vector(0,0,-1), ui.theme.icons.forward, true)
+	showDirection = displayDirectionalMarker(Vector(0,0,1), ui.theme.icons.backward, showDirection)
+	showDirection = displayDirectionalMarker(Vector(0,1,0), ui.theme.icons.up, showDirection, angle(forward, ui.pi))
+	showDirection = displayDirectionalMarker(Vector(0,-1,0), ui.theme.icons.down, showDirection, angle(forward, 0))
+	showDirection = displayDirectionalMarker(Vector(1,0,0), ui.theme.icons.right, showDirection)
+	showDirection = displayDirectionalMarker(Vector(-1,0,0), ui.theme.icons.left, showDirection)
+
+	if showDirection then
+		ui.addIcon(center, ui.theme.icons.direction_forward, ui.theme.colors.reticuleCircle, 32, ui.anchor.center, ui.anchor.center, nil, angle(forward, 0))
+	end
+
+end
+
+-- heading, pitch and roll
+local function displayReticulePitchHorizonCompass(center)
+	local heading, pitch, roll = Game.player:GetHeadingPitchRoll("planet")
+	local pitch_degrees = (pitch / ui.twoPi * 360)
+	local heading_degrees = (heading / ui.twoPi * 360)
+	local roll_degrees = (roll / ui.twoPi * 360);
+	local size = 12
+
+	if showNavigationalNumbers then
+		local uiPos = ui.pointOnClock(center, reticuleCircleRadius - size * 4, 3)
+		ui.addStyledText(uiPos, math.floor(pitch_degrees + 0.5) .. "°", ui.theme.colors.reticuleCircle, ui.fonts.pionillium.small, ui.anchor.right, ui.anchor.center, "Current pitch")
+
+		local uiPos = ui.pointOnClock(center, reticuleCircleRadius - size * 2, 0)
+		ui.addStyledText(uiPos, math.floor(heading_degrees + 0.5) .. "°", ui.theme.colors.reticuleCircle, ui.fonts.pionillium.small, ui.anchor.center, ui.anchor.top, "Current heading")
+
+		local uiPos = ui.pointOnClock(center, reticuleCircleRadius, 6)
+		ui.addStyledText(uiPos, math.floor(roll_degrees + 0.5) .. "°", ui.theme.colors.reticuleCircle, ui.fonts.pionillium.small, ui.anchor.center, ui.anchor.top, "Current roll")
+	end
+
+	displayReticulePitch(center, pitch_degrees)
+	displayReticuleHorizon(center, roll_degrees)
+	displayReticuleCompass(center, heading_degrees)
+end
+
+-- show the small indicator inside the reticule circle
+local function displayIndicator(center, onscreen, position, direction, icon, color, showIndicator)
+	local size = 32
+	local iconSize = 16
+	local dir = direction * reticuleCircleRadius * 0.90
+	local indicator = center + dir
+	if onscreen then
+		ui.addIcon(position, icon, color, size, ui.anchor.center, ui.anchor.center)
+	end
+	if showIndicator and (center - position):magnitude() > reticuleCircleRadius * 1.2 then
+		ui.addIcon(indicator, icon, color, iconSize, ui.anchor.center, ui.anchor.center)
+	end
+end
+
+local reticuleTarget = "frame"
+
+local function displayTargetData(target, center, radius, combatTarget, navTarget, colorLight, colorDark, frameColor, navTargetColor, combatTargetColor)
```

Comment:
This function is horribly long and undocumented. I'm guessing this displays the informations on the right side of the reticule. But they're not always related to the target (as the player understands the notion), since it can display info related to the current frame, right?

If I'm right, this needs at least a renaming, the current frame isn't a target, it's just a navigation context. AND COMMENTS !